### PR TITLE
feat: support isoclasses for 5 and 6 vertex undirected graphs

### PR DESCRIPTION
**Pull request by szhorvat**
See forum for details: https://igraph.discourse.group/t/isoclasses-and-motif-finding-for-5-and-6-vertex-graphs/1050

Changes in this PR:

 - Remove `isoclasses` arrays. They were not used, and no one remembers what they were for.
 - Document how the lookup tables are structured.
 - Add support for 5- and 6-vertex undirected graphs.

### CODE REVIEW

**Comment by szhorvat** [comment]
This is ready for general feedback, as it seems to be working correctly now, but I'll still make a few changes to simplify test code. The main test, which was modified here, will use a smaller graph so reduce the running time (LAD is slow at these motif sizes).

**Comment by szhorvat** [comment]
There's the question of where to put the code I used to create the lookup tables. It's written in Mathematica. We should keep it, otherwise it will be a pain to re-do everything from scratch next time we need to touch this functionality. I was hoping to make it short enough to insert it in comments, but that didn't quite work out.

**Comment by ntamas** [comment]
> There's the question of where to put the code I used to create the lookup tables.

Put them in `tools`; this is where we keep the LAPACK patches (I doubt that they still work, though) and an ARPACK-related `sed` script.

**Comment by szhorvat** [comment]
@ntamas When you review this, please read the comment I added at the top of `isoclasses.c` and let me know if it is understandable. I'll add the lookup table generator a bit later. Let me know if you're happy with this and I'll do the merge (including the merge into develop). Please don't "squash-and-merge". The commits here are many, but clean.

**Comment by szhorvat** [comment]
The reason for the reduced coverage in LAD is that I reduced the graph sizes in the `example::igraph_subisomorphic_lad` test. This test was already the slowest. With the larger motifs I added, it now takes 3.5 seconds when compiled with all optimizations.

File: src/isomorphism/isoclasses.c [hunk 1]
```diff
@@ -825,62 +2755,104 @@ int igraph_isoclass_create(igraph_t *graph, igraph_integer_t size,
...
-
     IGRAPH_VECTOR_INIT_FINALLY(&edges, 0);
 
 #define CHECK_ISOCLASS(number, directed, size) \
     IGRAPH_ERRORF( \
-        "Isoclass " IGRAPH_PRId " requested, but there are only " \
-        IGRAPH_PRId " %s graphs of size " IGRAPH_PRId ".", IGRAPH_EINVAL, \
+        "Isoclass %" IGRAPH_PRId " requested, but there are only %" \
+        IGRAPH_PRId " %s graphs of size %" IGRAPH_PRId ".", IGRAPH_EINVAL, \
         (igraph_integer_t) number, directed ? "directed" : "undirected", (igraph_integer_t) size)
```
Comment (ntamas) [inline src/isomorphism/isoclasses.c]: The total number of graphs of a given size is missing from the argument list of `IGRAPH_ERRORF`.

File: src/isomorphism/isoclasses.c [hunk 2]
```diff
@@ -27,355 +27,60 @@
...
+ * class is stored in "isographs".
+ *
+ * In the names of arrays, the number refers to the vertex count, while "u"
+ * indicates undirected graphs (the other arrays store directed ones).
+ *
+ * Description of each array for graphs of size n:
+ *
+ * isosclass_idx represents an n-by-n matrix stored in column-major order.
+ * Element i,j of the matrix is an integer with a single bit set. This bit,
+ * if set, represents edge i-j in the grah code.
```
Comment (ntamas) [inline src/isomorphism/isoclasses.c]: Typo here: `grah` -> `graph`

**Review by ntamas — CHANGES_REQUESTED** [review]
Looks good to me in general, but I have to trust you that the generated tables are correct :) Can you also add the Mathematica code that generated the tables?

**Comment by szhorvat** [comment]
@ntamas These two are fixed. I'll refrain from force-pushed while you're reviewing, but I'll clean up the commits afterwards.

**Review by ntamas — APPROVED** [review]

**Comment by ntamas** [comment]
Feel free to force-push and then merge, it seems okay to me now!

**Comment by szhorvat** [comment]
@ntamas Can you merge `master` into `develop` _before_ I merge this, so I can avoid any plfit-related merge conflicts when I later work on merging this into develop?

**Pull request merged**